src/new-components/checkbox-field/checkbox-field.tsx

```diff
@@ -104,7 +104,7 @@ const CheckboxField = React.forwardRef<HTMLInputElement, CheckboxFieldProps>(fun
                 indeterminate={indeterminate}
                 aria-hidden
             />
-            {label ? <Text>{label}</Text> : null}
+            {label ? typeof label === 'string' ? <Text>{label}</Text> : label : null}
```

Comment:
I think we need to put it inside `<Text />` regardless. Otherwise, it will not pick up the correct text color, for instance, when in dark mode.

I'm approving the PR, but with the expectation that this needs to be fixed before merging.

Reply:
But since this now supports _any_ `ReactNode`, we can pass along anything, won't the `<Text>` conflict with more complex labels? Although I don't see why one would do that, but capability is there.

I'm wondering if we should add a `startIcon` instead, like the `<Button>`. What do you think? We'll need a way to control the margin between the checkbox / icon / label, but I guess we can do that from the consumer side with proper selectors.

Reply:
I'm not a fan of a start icon. Where does it end? Will we need an end icon too? And if we want the icon, or any other markup in the middle? For instance, what if we want this:

```jsx
<CheckboxField
  label={<>This is <strong>important</strong> so pay attention}
/>
```

The labels in all the other fields already support `React.ReactNode`, this was an oversight with the checkbox label.

> But since this now supports _any_ `ReactNode`, we can pass along anything, won't the `<Text>` conflict with more complex labels? Although I don't see why one would do that, but capability is there.

The capability is there, but the primary goal of the label is to display text, and `<Text />` is what ensures that this text plays nicely with the design system theming and styles. You can see in `BaseField` (the component that serves as the basis for all fields except the checkbox) that we already wrap all the label are inside `<Text />`.

Reply:
That makes sense, thanks for clarifying.